### PR TITLE
i/o polish

### PR DESCRIPTION
**Pull request by morethanreal**
write docs, add aria-hidden, use iron-resizable, fix tests

PTAL @cdata cc @notwaldorf 

I documented the styling properties in `<iron-overlay-backdrop>` with a table. WDYT about adding docs in that style for elements that uses custom properties?

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -21,8 +21,9 @@
     "url": "git://github.com/PolymerElements/iron-overlay-behavior.git"
   },
   "dependencies": {
+    "polymer": "Polymer/polymer#^0.9.0",
     "iron-fit-behavior": "PolymerElements/iron-fit-behavior#^0.9.0",
-    "polymer": "Polymer/polymer#^0.9.0"
+    "iron-resizable-behavior": "PolymerElements/iron-resizable-behavior#^0.9.0"
```
Comment (cdata) [inline bower.json]: Woot. Did it work for you?

File: iron-overlay-backdrop.html [hunk 2]
```diff
@@ -9,11 +9,22 @@
...
+### Styling
+
+The following custom properties and mixins are available for styling.
+
+Custom property | Description | Default
+-------------------------------------------|------------------------|---------
+`--iron-overlay-backdrop-background-color` | Backdrop background color                                     | #000
+`--iron-overlay-backdrop-opacity`          | Backdrop opacity                                              | 0.6
+`--iron-overlay-backdrop`                  | Mixin applied to `iron-overlay-backdrop`.                      | {}
+`--iron-overlay-backdrop-opened`           | Mixin applied to `iron-overlay-backdrop` when it is displayed | {}
```
Comment (cdata) [inline iron-overlay-backdrop.html]: For some reason Github decided to highlight your table blood red.

File: iron-overlay-behavior.html [hunk 3]
```diff
@@ -10,44 +10,52 @@
...
+### Limitations
+
+The element is styled to appear on top of other content by setting its `z-index` property. You
+must ensure no element has a stacking context with a higher `z-index` than its parent stacking
+context. You should place this element as a child of `<body>` whenever possible.
+
+@polymerBehavior Polymer.IronOverlayBehavior
+*/
+
+  Polymer.IronOverlayBehaviorImpl = {
```
Comment (cdata) [inline iron-overlay-behavior.html]: Should we make `Impl` suffix a convention for this type of thing?

File: test/iron-overlay-behavior.html [hunk 4]
```diff
@@ -254,7 +254,8 @@
         test('backdrop is removed when the element is removed from DOM', function(done) {
           runAfterOpen(overlays[0], function() {
             var backdrop = overlays[0].backdropElement;
-            overlays[0].remove();
+            Polymer.dom(backdrop.parentNode).removeChild(backdrop);
+            Polymer.dom.flush();
```
Comment (cdata) [inline test/iron-overlay-behavior.html]: What does this do?

**Comment by cdata** [comment]
Minor questions, otherwise LGTM :+1:

File: bower.json (hunk 1 above)
Comment (morethanreal) [inline bower.json]: Seems good!

File: iron-overlay-behavior.html (hunk 3 above)
Comment (morethanreal) [inline iron-overlay-behavior.html]: Sure, if that means I don't have to change this :)

File: test/iron-overlay-behavior.html (hunk 4 above)
Comment (morethanreal) [inline test/iron-overlay-behavior.html]: I think Polymer.dom.removeChild (and other DOM operations) is sometimes lazy, and flush causes it to be sync.

**Pull request merged**